types/react-native/index.d.ts

```diff
@@ -3515,7 +3515,7 @@ export interface FlatListProperties<ItemT> extends VirtualizedListProperties<Ite
     /**
      * Rendered in between each item, but not at the top or bottom
      */
-    ItemSeparatorComponent?: React.ComponentType<any> | null
+    ItemSeparatorComponent?: React.ComponentType<any> | (() => React.ReactElement<any>) | null
```

Comment:
I think this one already supported a function, because that’s what the `React.ComponentType` type is for. So I think you only need to change the `React.ComponentClass` instances below to use `React.ComponentType` as well.

Reply:
oh, thank you. I'll try this method.